feat(kit): new pipe `HintErrorPipe`

vladimirpotekhin opened the pull request:
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes


Closes #1399 


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

splincode commented on projects/kit/pipes/hint-error/hint-error-pipe.ts:
```diff
@@ -0,0 +1,35 @@
+    pure: false,
+})
+export class TuiHintErrorPipe implements PipeTransform, OnDestroy {
+    asyncPipe!: AsyncPipe;
+    fieldErrorPipe!: TuiFieldErrorPipe;
```
private?

waterplea commented on projects/kit/pipes/hint-error/hint-error-pipe.ts:
```diff
@@ -0,0 +1,34 @@
+    private readonly fieldErrorPipe!: TuiFieldErrorPipe;
+
+    constructor(@Inject(INJECTOR) readonly injector: Injector) {
+        const localInjector = Injector.create({
```
Let's move this to private fields out of constructor. Also `injector` should be `private` and `INJECTOR` is only needed if you do not use static `Injector.create` which we do, so you might as well change it to `Injector`.

waterplea commented on projects/kit/pipes/hint-error/hint-error-pipe.ts:
```diff
@@ -0,0 +1,34 @@
+        this.fieldErrorPipe = localInjector.get(TuiFieldErrorPipe);
+    }
+
+    transform(order: ReadonlyArray<string>): PolymorpheusContent | null {
```
Let's drop `null`

waterplea reviewed (commented):
How about we rename it to `tuiFieldErrorContent` pipe and put it in the same module?

vladimirpotekhin commented:
> How about we rename it to `tuiFieldErrorContent` pipe and put it in the same module?

Good idea 👍 
Renamed and moved in `TuiFieldErrorPipeModule`
